src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc

```diff
@@ -201,16 +202,16 @@ void ReplayStatusFormatter<I>::handle_update_tag_cache(uint64_t master_tag_tid,
       tag_data.predecessor.mirror_uuid !=
         librbd::Journal<>::ORPHAN_MIRROR_UUID) {
     dout(20) << "hit remote image non-primary epoch" << dendl;
-    tag_data.predecessor.tag_tid = mirror_tag_tid;
+    tag_data.predecessor = {"", true, mirror_tag_tid, 0};
```

Comment:
Nit: might as well just set the ```tag_tid``` to zero since you cannot make any progress in the decode loop: ```tag_data.predecessor = {}```

Reply:
Thanks, updated. I also had to update send_update_tag_cache to test for `master_tag_tid == 0` as a termination condition.